tests/test_canvas_object.py

```diff
@@ -62,3 +62,22 @@ def test_set_attributes_invalid_date(self, m):
         self.assertFalse(hasattr(self.canvas_object, "end_at_date"))
         self.assertTrue(hasattr(self.canvas_object, "start_at"))
         self.assertTrue(hasattr(self.canvas_object, "end_at"))
+
+    # set_attributes 'content-type'
+    def test_set_attributes_with_content_type(self, m):
+        attributes = {
+            "content-type": "application/json",
+            "content_type": "another_application/json",
+            "filename": "example.json",
+        }
+
+        self.canvas_object.set_attributes(attributes)
+
+        self.assertTrue(hasattr(self.canvas_object, "content-type"))
+        self.assertEqual(
+            getattr(self.canvas_object, "content-type"), "application/json"
+        )
+        self.assertTrue(hasattr(self.canvas_object, "content_type"))
+        self.assertEqual(self.canvas_object.content_type, "another_application/json")
```

Comment:
This test brings up some bigger questions about how we want to handle the case where both the `-` and `_` versions of what is otherwise the same variable are set. In this case, the order of the items in the dictionary matters. I'll create and push new a test case the same as this one but reversing the order to demonstrate.

It's pretty unlikely that Canvas would send back two attributes with only the `-`/`_` differentiating them, let alone with different values, but it's something we'll need to consider how we want to handle.

Reply:
Yeah, I thought about this too, I think we can solve both of those problems but maybe in a separate issue. I don't think it's super likely either.

Reply:
Would it be worth adding tests that detect if Canvas would ever do this?

I do quite some API "reverse engineering" of APIs I want to use but don't have any official documentation (they're for "internal" use of the web UI). Then I add tests that will fail whenever some return value from the server changes.

Such tests must be added as a separate test suite run for this particular purpose. It would require a Canvas instance to run them against and, consequently, a login and token to the API. So not the ordinary test setup.

I'm leaning towards not worth it. Canvas is pretty stable and documents everything. So the likelihood of being useful is close to zero.